### ISSUE TITLE
EmrCreateJobFlowOperator should not wait_for_completion if wait_for_completion=false deferrable=true

### ISSUE THREAD

**Issue by ibzx** [apache/airflow#40966]
### Apache Airflow version

2.9.3

### What happened?

When setting global 
```
AIRFLOW__OPERATORS__DEFAULT_DEFERRABLE =true
EmrCreateJobFlowOperator(wait_for_completion=false)
```
The operator should adds a deferral trigger to wait for Jobflow create completion status.

### What you think should happen instead?

Since we did not ask to wait for completion, the deferrable argument should do nothing. The deferrable argument should only start a trigger when `wait_for_completion=true`

### How to reproduce

See deferrable code here:
https://github.com/apache/airflow/blob/460d2095cda6674dda4860d605573b8f1cec7de5/airflow/providers/amazon/aws/operators/emr.py#L827-L849

Sample code
```
EmrCreateJobFlowOperator(
        task_id="cluster_m5",
        wait_for_completion=false,
        deferrable=true
)
```

### Operating System

Mac OS 14.5 (23F79)

### Deployment

Astronomer

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by Prab-27** [comment]
I am using WSL2 set up can I work on this issue here OS is Mac ? .If yes , I would like to work on this issue as my first contribution.

**Comment by laksh-krishna-sharma** [comment]
May I work on this issue with a little bit of your guidance?

**Comment by eladkal** [comment]
cc @vincbeck I think you fixed similar problem for emr serverless?

**Comment by vincbeck** [comment]
Correct. It seems many operators have the same issue. This is a very good first issue so please feel free to work on it and tag me to review it :) Here are some examples: #41191, #41103, #41059

**Comment by vincbeck** [comment]
@laksh-krishna-sharma Please look at the example I provided, that should help you to fix this issue

**Comment by laksh-krishna-sharma** [comment]
Thank you, @vincbeck . I will start working on it immediately. I will reach out if I have any questions.

**Comment by dominikhei** [comment]
@laksh-krishna-sharma Do you want to finish your PR? I can give you some guidance with the remaining test cases that failed, if you want to

**Comment by laksh-krishna-sharma** [comment]
sure sir @dominikhei

**Comment by laksh-krishna-sharma** [comment]
Hi @dominikhei, may I work on this issue again? I’ve gained more experience since last time and feel confident I can handle it now
